### PR TITLE
test(rln): Implement rln tests

### PR DESCRIPTION
**Pull request by AlejandroCabeza**
# Description
Extend RLN tests coverage

# Changes
- [ ] Implement test plan

### CODE REVIEW

File: tests/waku_rln_relay/test_rln_group_manager_onchain.nim [hunk 1]
```diff
@@ -257,6 +70,13 @@ suite "Onchain group manager":
     (await manager2.init()).isErrOr:
       raiseAssert "Expected error when contract address doesn't match"
 
+  asyncTest "should error if contract does not exist":
+    let manager = await setup()
+    manager.ethContractAddress = "0x0000000000000000000000000000000000000000"
+
+    expect(CatchableError):
+      await manager.init()
+
```
Comment (rymnc) [inline tests/waku_rln_relay/test_rln_group_manager_onchain.nim:79]: this test already exists, i believe

File: tests/node/test_wakunode_sharding.nim [hunk 2]
```diff
@@ -431,7 +432,7 @@ suite "Sharding":
         let clientResult = await clientHandler.waitForResult(FUTURE_TIMEOUT)
         assertResultOk(clientResult)
 
-      xasyncTest "store (automatic sharding filtering)":
+      asyncTest "store (automatic sharding filtering)":
```
Comment (rymnc) [inline tests/node/test_wakunode_sharding.nim:435]: should we split this change into a different PR?

File: tests/waku_keystore/utils.nim [hunk 3]
```diff
@@ -0,0 +1,31 @@
+{.used.}
+
+when (NimMajor, NimMinor) < (1, 4):
+  {.push raises: [Defect].}
+else:
+  {.push raises: [].}
```
Comment (rymnc) [inline tests/waku_keystore/utils.nim:6]: I don't think we need this in our files anymore :)
cc: @Ivansete-status for confirmation

File: tests/waku_rln_relay/test_rln_group_manager_onchain.nim [hunk 4]
```diff
@@ -285,6 +92,19 @@ suite "Onchain group manager":
     (await manager2.init()).isErrOr:
       raiseAssert "Expected error when contract address doesn't match"
 
+  # FAILED
```
Comment (rymnc) [inline tests/waku_rln_relay/test_rln_group_manager_onchain.nim:95]: is this the test that's failing now?

File: tests/waku_rln_relay/test_wakunode_rln_relay.nim [hunk 5]
```diff
@@ -11,12 +11,35 @@ import
...
 import
   waku/[waku_core, waku_node, waku_rln_relay],
-  ../testlib/wakucore,
-  ../testlib/wakunode,
+  ../testlib/[wakucore, futures, wakunode],
   ./rln/waku_rln_relay_utils
 
 from std/times import epochTime
 
+proc buildWakuRlnConfig_versionAware(
```
Comment (rymnc) [inline tests/waku_rln_relay/test_wakunode_rln_relay.nim:19]: don't think we need it to be versionAware now :)
```suggestion
proc buildWakuRlnConfig(
```

File: tests/waku_rln_relay/test_wakunode_rln_relay.nim [hunk 6]
```diff
@@ -11,12 +11,35 @@ import
...
+    credIndex: uint,
+    epochSizeSec: uint64,
+    treeFilename: string,
+    userMessageLimit: int = 1,
+): WakuRlnConfig =
+  let treePath = genTempPath("rln_tree", treeFilename)
+  # Off-chain
+  when defined(rln_v2):
```
Comment (rymnc) [inline tests/waku_rln_relay/test_wakunode_rln_relay.nim:27]: we only use rln_v2 now, so this compiler directive can be removed

File: tests/waku_rln_relay/utils_onchain.nim [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+
+import
+  waku/[
+    waku_rln_relay/protocol_types,
+    waku_rln_relay/constants,
+    waku_rln_relay/contract,
+    waku_rln_relay/rln,
+    waku_rln_relay/conversion_utils,
+    waku_rln_relay/group_manager/on_chain/group_manager
+  ],
```
Comment (rymnc) [inline tests/waku_rln_relay/utils_onchain.nim:28]: a lot of this is exported by the top level module
`waku/waku_rln_relay`

**Review by rymnc — COMMENTED** [review]
Left some comments in-line! 
This PR has some awesome QoL changes, apologies for the delay in getting it merged!!

File: tests/waku_keystore/utils.nim (hunk 3 above)
Comment (Ivansete-status) [reply tests/waku_keystore/utils.nim:6]: Exactly @rymnc , only ` {.push raises: [].}`

File: tests/waku_rln_relay/test_rln_group_manager_onchain.nim (hunk 1 above)
Comment (AlejandroCabeza) [reply tests/waku_rln_relay/test_rln_group_manager_onchain.nim]: Will double check.

File: tests/waku_keystore/utils.nim (hunk 3 above)
Comment (AlejandroCabeza) [reply tests/waku_keystore/utils.nim:6]: Alrighty :)

File: tests/waku_rln_relay/test_rln_group_manager_onchain.nim (hunk 4 above)
Comment (AlejandroCabeza) [reply tests/waku_rln_relay/test_rln_group_manager_onchain.nim:95]: No, sorry. That was a cue for myself which I intended to delete.

File: tests/waku_rln_relay/test_wakunode_rln_relay.nim (hunk 5 above)
Comment (AlejandroCabeza) [reply tests/waku_rln_relay/test_wakunode_rln_relay.nim:19]: True, thanks for the heads up.

File: tests/waku_rln_relay/test_wakunode_rln_relay.nim (hunk 6 above)
Comment (AlejandroCabeza) [reply tests/waku_rln_relay/test_wakunode_rln_relay.nim:27]: Great!

File: tests/node/test_wakunode_sharding.nim (hunk 2 above)
Comment (AlejandroCabeza) [reply tests/node/test_wakunode_sharding.nim:435]: I don't mind either way. If you rather have it separate I'll move it.

File: tests/waku_rln_relay/test_rln_group_manager_onchain.nim (hunk 1 above)
Comment (AlejandroCabeza) [reply tests/waku_rln_relay/test_rln_group_manager_onchain.nim]: I couldn't find it. 
Perhaps you're referring to "should error on initialization when loaded metadata does not match". I did a quick check on that one and the error message is not the same as the one in this test.

**Review by rymnc — APPROVED** [review]
LGTM, checks are failing though, is it due to an unrelated change?

**Comment by AlejandroCabeza** [comment]
> LGTM, checks are failing though, is it due to an unrelated change?

I think they are failing in master as well.

@Ivansete-status Can you confirm?

**Review by Ivansete-status — APPROVED** [review]

**Comment by Ivansete-status** [comment]
> > LGTM, checks are failing though, is it due to an unrelated change?
> 
> I think they are failing in master as well.
> 
> @Ivansete-status Can you confirm?

Yes you are absolutely right, this issue also happens in `master`.
